### ISSUE TITLE
Numeric data in bucket_entity_data

### ISSUE THREAD

**Issue by Mineralium** [Estecka/mc-Enchants-CIT#18]
I am doing a personal project with tropical fish variants, but it doesn't work and although I have gone through all available 2.6 documentations, I can't figure out what I am doing wrong.
My module file looks like this:
```
{
	"type": "bucket_entity_data",
	"items": ["tropical_fish_bucket"],
	"modelPrefix": "item/tfb_textures/",
	"modelParent": "item/generated",
	"parameters": {
		"nbtPath": "BucketVariantTag"
	}
}
```
My best guess is that I misunderstood what "nbtPath" does, but there isn't really much other data...
Tropical fish nbt goes like this:
```
tropical_fish_bucket[minecraft:bucket_entity_data={"BucketVariantTag":0,"Health":3.0}]
```
The "0" being the variant. Thus the custom texture is called "0.png" and is located in the "tfb_textures" folder.
So.. please help, and also maybe make the documentation more clear.

Btw, I know 2.7 is now available, but I am using 1.21 which doesn't have the required Fabric API version.

**Comment by Estecka** [comment]
This is not on you, nbt_data modules completely ignore non-string data.

Is this nbt the vanilla format for tropical fish ? If so I can make a small adjustment to allow for numeric data to be converted.

> Btw, I know 2.7 is now available, but I am using 1.21 which doesn't have the required Fabric API version.

What do you mean ? v2.6 and v2.7 should have the same dependencies.

**Comment by Estecka** [comment]
Had a look at how tropical fish work, and it turns out `BucketVariantTag` contains both the fish's pattern *and* its two colors. Unless you were planning on creating a thousand separate models, just getting the pattern will require a more specialised module, similarly to `axolotl_variant`.

Did you plan on taking the fish's colours into account ? I don't know of a convenient way to do that in MC 1.21.0, but MC 1.21.4 might make it easier with the tint source options in the new item states.

**Comment by Mineralium** [comment]
Sorry about the dependencies thing. I forgot I myself modified the 2.6 version to support older Fabric API, lol. I still tested on 1.21.1 though to make sure nothing breaks.

This mod is updated faster than CIT Resewn, so I am pretty much updating all CIT packs I use to this format. And since I am obviously doing it without permission, and with no intent to share, I call it a "personal project". So basically someone else already made the thousands of textures.

**Comment by Estecka** [comment]
Here's a build that should accept numeric data like you originally expected, let me know if this works for your use case before I release it: https://github.com/Estecka/mc-Variants-CIT/actions/runs/12215313844

I also lowered the dependency on FAPI, since it seems to work with 0.102

**Comment by Mineralium** [comment]
So.. this is a bit hard to explain. When I tested with a single texture, everything worked perfectly. When I added the rest of ~3000, everything absolutely broke. And when I reduced back to a single texture, it was still broken. Reinstalling also didn't fix it. Will do more testing.

But all other packs utilizing this mod still work, even my recreating of ItemBounds, so it seems contained to tropical fish buckets.

Edit: By broken I mean random textures are applied. Interestingly I have a similar bug with connected glass by Continuity, but that is fixed by simply doing f3+t while this isn't.

**Comment by Estecka** [comment]
I tried using your module in [All The Fish](https://modrinth.com/resourcepack/all-the-fish), and things worked as I expected. Althought there were a few pitfalls to avoid:
- Some fish have the wrong shape, but that's an issue with the pack's textures being misnamed. Is this what you meant by random textures ?
- The textures needed to be moved to the `minecraft` namespace. Because numbers are not namespaced, they will always default to this.
- The pack used `item/xxx.png` as its naming scheme, which is way too vague. If you changed your model prefix to match the pack's, that could very easily conflict with other packs; moving them to the subfolder `tfb_textures/` was a good call.

Otherwise, try disabling other CIT or optimization mods that could be interfering. ModernFix in particular has a track record of breaking things.

**Comment by Mineralium** [comment]
Never mind! It works well, thank you!

For some reason the textures I used got corrupted when I copied them over. Maybe because of the amount, or maybe some other reason, but it isn't the mod's fault. Sorry, should have tested more, but was a bit tired.

Just want to add that there are other data in "entity_data" which requires number values. Like "RabbitType" and "Size", but I have never seen anyone utilize those for CIT.
